### ISSUE TITLE
Accessibility for Colourblind Individuals

### ISSUE THREAD

**Issue by kbpickens** [NeunEinser/bingo#109]
While watching a streamer playing, I noticed and others discussed the difficulty of seeing the difference between found and not found items for colourblind individuals.  Can you implement a pattern change to the backgrounds in addition to the colour change?  Perhaps, place a black and white line across the cells behind the item (I'm basing this suggestion on what I did for nitwits in a personal resource pack).

**Comment by Kri5t0fK** [comment]
1. One (quite easy) option - change `cell backgrounds` in `default background` with **full/framing pattern**
Some suggestions below:
![image](https://user-images.githubusercontent.com/53398204/151045932-593338a5-3732-4942-a143-dbcd9985119d.png)
Of course, it only adds pattern for _unobtained_ items, and determining teams (that _obtained_ the item) is based on `team snips` being in particular angles.

2. Another option - leave `default background` as it is, add **patterns** to all snips in [/textures/font/completed_item.png](https://github.com/NeunEinser/bingo/blob/master/assets/bingo/textures/font/completed_item.png)
But in this case, there would probably need to be done 16 (for each team) unique patterns on snips consisting of just a few pixels.

**Comment by NeunEinser** [comment]
Yeah, I feel like the best option would be to have unique patterns per team color, but yeah that'd be quite some afford as there would need be a lot more textures for each team. I also don't know how recognizable it would actually be with smaller tiles.

For now, I am thinking to change the default background to white, so that there is hopefully always contrast in terms of brightness for completed squares, and possibly add a border to the tiles.

**Comment by NeunEinser** [comment]
To list some further ideas here (which are out of scope for 5.0 release unfortunately though).
Since in multiplayer you can have multiple colors on the card at once, I think the problem is much bigger than simply make one color work.

- One unique pattern per team. Should be repeated over the entire tile. Possibly no pattern by default and "color blind mode" preference setting. The pattern should solely rely on "brightness" of the color, and not the hue. The pattern must be easily recognizable in grayscale.
- Try to rearrange colors to avoid having similar colors or colors where people with the most common color deficiencies have issues
- Have colors that are next to each other have different base "brightness" values.
- Make sure all color's brightness value is sufficiently different from the card background that there is an easily recognizable difference in grayscale.
- Possibly some sort of border around each color shape.

Another problem I see is color deficient people not knowing a person of which team just got an item. Player names are colored by team color, but that doesn't help much here of course.
The idea to solve this problem is to use the unique pattern for the team prefix. This requires a custom font, but I think you can use JSON text components for team prefixes, so that shouldn't be an issue. The next best thing is to simply modify the "player got item" message to include the team pattern, but if team prefix works, it also shows up when people send regular chat messages.
The pattern should also be included in join messages like "<player> joined the <pattern icon> red team".
If "color blind mode" is a preference setting, this should possibly not be shown for regular people, though I do not see how this would be possible with team prefixes as they are global.

It would be desired to find and talk to color deficient people from the community to discuss and iterate on ideas and wip screenshots, preferably talking to people having different kinds of deficiencies.
@kbpickens if you have contact of people who are affected, or are affected yourself, you might be able to help with this once work on this starts.

**Comment by kbpickens** [comment]
I am affected (officially, incomplete colorblindness), and would be glad to help.
